src/unit_test.h

```diff
@@ -0,0 +1,138 @@
+/***********************************************************************
+ * Copyright (c) 2025  Matias Furszyfer (furszy)                       *
+ * Distributed under the MIT software license, see the accompanying    *
+ * file COPYING or https://www.opensource.org/licenses/mit-license.php.*
+ ***********************************************************************/
+
+#ifndef LIBSECP256K1_UNIT_TEST_H
+#define LIBSECP256K1_UNIT_TEST_H
+
+/* --------------------------------------------------------- */
+/* Configurable constants                                    */
+/* --------------------------------------------------------- */
+
+/* Maximum number of command-line arguments.
+ * Must be at least as large as the total number of tests
+ * to allow specifying all tests individually. */
+#define MAX_ARGS 150
+/* Maximum number of parallel jobs */
+#define MAX_SUBPROCESSES 16
+
+/* --------------------------------------------------------- */
+/* Test Framework Registry Macros                            */
+/* --------------------------------------------------------- */
+
+#define CASE(name) { #name, run_##name }
+#define CASE1(name) { #name, name }
```

Comment:
6162b7617d3731e08d2b59baf79ba602182cf85e:

It's unfortunate to have such a pair of confusing macros. Perhaps their naming could be improved, although I don’t have a concrete suggestion.